src/Semantic/Graph.hs

```diff
@@ -106,35 +121,44 @@ runGraph :: ( Member Distribute sig
          -> Bool
          -> Project
          -> m (Graph ControlFlowVertex)
-runGraph ImportGraph _ project
-  | SomeAnalysisParser parser (lang' :: Proxy lang) <- someAnalysisParser (Proxy :: Proxy AnalysisClasses) (projectLanguage project) = do
-    let parse = if projectLanguage project == Language.Python then parsePythonPackage parser else fmap (fmap snd) . parsePackage parser
-    package <- parse project
-    runImportGraphToModuleInfos lang' package
-runGraph CallGraph includePackages project
-  | SomeAnalysisParser parser lang <- someAnalysisParser (Proxy :: Proxy AnalysisClasses) (projectLanguage project) = do
-    let parse = if projectLanguage project == Language.Python then parsePythonPackage parser else fmap (fmap snd) . parsePackage parser
-    package <- parse project
-    modules <- topologicalSort <$> runImportGraphToModules lang package
-    runCallGraph lang includePackages modules package
-
-runCallGraph :: ( VertexDeclarationWithStrategy (VertexDeclarationStrategy syntax) syntax syntax
-                , Declarations1 syntax
-                , AccessControls1 syntax
-                , Ord1 syntax
-                , Functor syntax
-                , Evaluatable syntax
-                , term ~ Term syntax Loc
-                , FreeVariables1 syntax
+runGraph type' includePackages project
+  | Just (SomeParser parser) <- Map.lookup (projectLanguage project) analysisParsers
+  , SomeLanguage (lang :: Proxy lang) <- reifyLanguage (projectLanguage project) = do
+    package <- if projectLanguage project == Language.Python then
+        parsePythonPackage parser project
+      else
+        fmap snd <$> parsePackage parser project
+    case type' of
+      ImportGraph -> runImportGraphToModuleInfos lang package
+      CallGraph -> do
+        modules <- topologicalSort <$> runImportGraphToModules lang package
+        runCallGraph lang includePackages modules package
+  | otherwise = error $ "Analysis not supported for: " <> show (projectLanguage project)
+
+data SomeLanguage where
+  SomeLanguage :: HasPrelude lang => Proxy lang -> SomeLanguage
+
+reifyLanguage :: Language -> SomeLanguage
+reifyLanguage = \case
+  Go         -> SomeLanguage (Proxy @'Go)
+  JavaScript -> SomeLanguage (Proxy @'JavaScript)
+  PHP        -> SomeLanguage (Proxy @'PHP)
+  Python     -> SomeLanguage (Proxy @'Python)
+  Ruby       -> SomeLanguage (Proxy @'Ruby)
+  TypeScript -> SomeLanguage (Proxy @'TypeScript)
+  TSX        -> SomeLanguage (Proxy @'TSX)
+  l          -> error $ "HasPrelude not supported for: " <> show l
```

Comment:
`SomeAnalysisParser` also provided us with a witness for the `HasPrelude` instance, so we now define that locally.